internals/daemon/api_plan_test.go

```diff
@@ -181,20 +184,192 @@ services:
+func (s *apiSuite) TestLayersCombineOneError(c *C) {
+	testLayersCombineFormatError(s, c,
+		"services:\n dynamic:\n  command: echo dynamic\n",
+		`layer "base" must define "override" for service "dynamic"`)
+
+}
+
+func (s *apiSuite) TestLayersCombineServicesMultiErrors(c *C) {
+	input := `
+services:
+	svc1:
+	  override: replace
+	  command: ""
+	  on-success: invalid
+	  on-failure: invalid
+	  on-check-failure:
+		 svc1: foo
+	  backoff-factor: 0
+	svc2:
+	  override: replace
+	  command: invalidCommand
+	  on-success: invalid
+	  on-failure: invalid
+	  on-check-failure:
+		 svc1: foo
+	  backoff-factor: 0
+	svc3:
+	  override: foo
+	  command: ""
+	  on-success: invalid
+	  on-failure: invalid
+	  on-check-failure:
+		 svc3: foo
+	  backoff-factor: 0
+	svc4:
+	  command: cmd
+	  on-success: invalid
+	  on-failure: invalid
+	  on-check-failure:
+		 svc4: foo
+	  backoff-factor: 0
+	  `
+
+	expectedError := `multiple errors validating plan:
+- layer "base" has invalid "override" value for service "svc3"
+- layer "base" must define "override" for service "svc4"
+- plan service "svc2" on-success action "invalid" invalid
+- plan service "svc2" on-failure action "invalid" invalid
+- plan service "svc2" on-check-failure action "foo" invalid
+- plan service "svc2" backoff-factor must be 1.0 or greater, not 0
+- plan must define "command" for service "svc1"
+- plan service "svc1" on-success action "invalid" invalid
+- plan service "svc1" on-failure action "invalid" invalid
+- plan service "svc1" on-check-failure action "foo" invalid
+- plan service "svc1" backoff-factor must be 1.0 or greater, not 0
+`
+
+	testLayersCombineFormatError(s, c, input, expectedError)
+
+}
+
+func (s *apiSuite) TestLayersCombineChecksMultiErrors(c *C) {
+	input := `
+checks:
+  svc1:
+    override: replace
+    level: foo
+    period: 0
+    timeout: 0
+    http:
+      url: ""
+    tcp:
+      port: 0
+    exec:
+      command: ""
+      service-context: foo
+      user-id: 0
+  svc2:
+    override: replace
+    level: alive
+    period: 10s
+    timeout: 10s
+    exec:
+      command: invalid
+  svc3:
+    override: foo
+    level: alive
+    exec:
+      command: ""
+  svc4:
+    level: alive
+    exec:
+      command: cmd
+	  `
+
+	expectedError := `multiple errors validating plan:
+- layer "base" has invalid "override" value for check "svc3"
+- layer "base" must define "override" for check "svc4"
+- plan check "svc2" timeout must be less than period
+- plan check "svc1" level must be "alive" or "ready"
+- plan check "svc1" period must not be zero
+- plan check "svc1" timeout must not be zero
+- plan must set "url" for http check "svc1"
+- plan must set "port" for tcp check "svc1"
+- plan must set "command" for exec check "svc1"
+- plan check "svc1" service context specifies non-existent service "foo"
+- plan check "svc1" has invalid user/group: must specify group, not just UID
+- plan must specify one of "http", "tcp", or "exec" for check "svc1"
```

Comment:
nit: this message is a bit unclear, it should probably say that the check can **only** specify one of `http`, `tcp`, `exec`

Reply:
one more thing - if we get this message because more than one of `http`, `tcp`, `exec` is specified, then there's no need to validate those sections. So this error message should just have
```
plan must specify one of "http", "tcp", or "exec" for check "svc1"
```
and skip the validation for those sections, e.g. don't print
```
- plan must set "command" for exec check "svc1"
- plan check "svc1" service context specifies non-existent service "foo"
- plan check "svc1" has invalid user/group: must specify group, not just UID
```